src/components/try-hcp-callout/components/a-b-test.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react'
+import { isInUS } from '@hashicorp/platform-util/geo'
+import { useFlags } from 'flags/client'
+import { abTestTrack } from 'lib/ab-test-track'
+
+const ctaSuffix = 'for free'
+
+export function useABTestCta(ctaText: string) {
+	const { flags } = useFlags()
+	const flag = flags?.tryCloudFreeCta || false
+
+	let final = ctaText
+
+	if (isInUS() && flag && !ctaText.endsWith(ctaSuffix)) {
+		final = `${ctaText} ${ctaSuffix}`
+	}
+
+	useEffect(() => {
+		abTestTrack({
+			type: 'Served',
+			test_name: 'HCP CTA Trial 2023-02',
+			variant: flag.toString(),
+		})
```

Comment:
Need to validate if this is the correct way to handle a GM campaign. cc @alexcarpenter

Reply:
🤔 Thinking through whether I should add a plain 'click' track event for the CTA link, or if I use this function again with 'Result'. Open to feedback!

Reply:
I think I'll wait to hear back from the GM folks to see if they need more events and not add the 'result' yet. Because i'm sure we already have events with product intent firing to the cloud platform.